autoload/sj/yaml.vim

```diff
@@ -139,5 +139,87 @@ function! s:GetChildren(line_no)
   endwhile
   let next_line_no = next_line_no - 1
 
+  " Preserve trailing empty lines
+  while sj#BlankString(getline(next_line_no)) && next_line_no > line_no
+    let next_line_no = next_line_no - 1
+  endwhile
+
   return [sj#GetLines(line_no + 1, next_line_no), next_line_no]
 endfunction
+
+function! s:splitArrayLines(array)
+  let lines = []
+
+  let line = ''
+  let fences = { '"': '"', "'": "'", '{': '}' }
+
+  for chunk in split(a:array, ',')
+    if line != ''
+      " Inside a string
+      let line = line . ',' . chunk
+
+      if chunk =~ fences[line[0]] . '\s*$'
+        " End of string
+        call add(lines, s:convertToSingleLine(line))
+        let line  = ''
+      endif
+
+    " Start of string
+    elseif chunk =~ '^\s*[' . join(keys(fences), '') . ']'
+      let line  = sj#Ltrim(chunk)
+
+      if chunk =~ fences[line[0]] . '\s*$'
+        " Complete string
+        call add(lines, s:convertToSingleLine(line))
+        let line  = ''
+      endif
+    else
+      call add(lines, s:convertToSingleLine(chunk))
+    endif
+  endfor
+
+  if line != ''
+    cal add(lines, s:convertToSingleLine(line))
+  endif
+
+  return lines
+endfunction
+
+function! s:joinArrayLines(lines)
+  let lines = map(a:lines, 's:convertToCompactLine(v:val)')
+  return join(lines, ', ')
+endfunction
+
+function! s:convertToCompactLine(line)
+  let prop = substitute(a:line, '\v^([a-z]+):.*', '\1', 1)
```

Comment:
I don't quite understand how this works. For starters, I think the `1` at the end would be more commonly `"g"`? Does this substitution check if there is a key/value pair? Like, if the line gets its bits substituted, it would be different from the original, but could this logic just see if the regex matches instead?

Reply:
🙈I need to improve this, this really only works for keys containing lower case characters...

I should probably just reuse `s:splitKeyValue(line)`